core/chains/evm/headtracker/head_tracker.go

```diff
@@ -159,7 +159,7 @@ func (ht *headTracker) Backfill(ctx context.Context, headWithChain *evmtypes.Hea
 		baseHeight = 0
 	}
 
-	return ht.backfill(ctx, headWithChain.EarliestInChain(), baseHeight)
+	return ht.backfill(ctx, headWithChain.EarliestInChain().(*evmtypes.Head), baseHeight)
```

Comment:
Are we going to end up with typecasts everywhere like this now that head is more generic? That seems like a step backwards

Reply:
This was due to the fact that EarliestInChain() was already used by evm native code, and also needed for generic Txmgr, so i merged them to a single interface method, that returns interface Head, instead of native head.

Changed it now.
The interface method is different from native method, so avoids having to use different types.